src/Transport/Addressing/Composition/Strategies/HierarchyComposition.cs

```diff
@@ -25,10 +25,17 @@ public string GetEntityPath(string entityName, EntityType entityType)
             {
                 case EntityType.Queue:
                 case EntityType.Topic:
-                    return pathGenerator(entityName) + entityName;
+                    var prefix = pathGenerator(entityName);
+                    if (entityName.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return entityName;
+                    }
+                    return prefix + entityName;
```

Comment:
Not really required but would string interpolation be more readable?

Reply:
Same readability to be honest. 
I personally don't see much difference between `$"{prefix}{entityName}"` and `prefix + entityName`.
But will change to make your day better 😜